django/db/models/query.py

```diff
@@ -1619,8 +1620,19 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
 
         if single:
             val = vals[0] if vals else None
-            to_attr = to_attr if as_attr else cache_name
-            setattr(obj, to_attr, val)
+
+            if as_attr:
+                # A to_attr has been given for the prefetch
+                setattr(obj, to_attr, val)
+            elif is_descriptor:
+                # cache_name points to a field name in obj
+                # this field is likely a descriptor for a related object.
```

Comment:
if we are confident enough to name the variable `is_descriptor`, then this comment should be updated to make a stronger assertion?